src/Pandell.PackageRunner/Program.cs

```diff
@@ -215,17 +252,29 @@ private int Run(Assembly packageRunnerAssembly, string packageRunnerExeFileName,
             using (var catalog = new AggregateCatalog(assemblies))
             using (var container = new CompositionContainer(catalog))
             {
+                Program.WriteLine(verbose, "Resolving execution delegate.");
                 container.SatisfyImportsOnce(this);
                 if (this.RunAssembly == null)
                 {
                     Console.Error.WriteLine("Method not found!");
                     return 5;
                 }
+                Program.WriteLine(verbose, "Executing delegate.");
                 this.RunAssembly(configuration.token);
+                Program.WriteLine(verbose, "Delegate execution done.");
                 return 0;
             }
         }
 
+        /// <summary>
+        /// </summary>
+        private static void WriteLine(bool outputActive, string message, object[] args = null)
+        {
+            if (outputActive)
+            {
+                Console.WriteLine(message, args);
```

Comment:
@milang , I did it but I have one problem with this. The delegate type should be referenced/imported in the package(s) using it. This is not very good as referencing the `exe` is not common practice and it would introduce one extra dependency. Using generic `Action` delegate have drawbacks as well. Because it doesn't support `params` and would make objects argument required. Probably the best signature would be `Action<TraceLevel, string>` and leave formatting to the caller. And this is what I changed it to. Let me know if you have a better idea.